### DISCUSSION TITLE
SSR Data Revalidation / Refresh

### DISCUSSION THREAD

**Discussion by kyjus25** [withastro/roadmap#595] [Proposal]
# Summary

Implement manual data refresh / revalidation, specifically in SSR mode

# Background & Motivation

It is difficult (or impossible?) to re-fetch server side data without reloading the browser or re-routing to the same page. For instance, when a user's name is updated via a form on the page, the `User` object as a whole should be re-fetched from the server that was shared via middleware `Astro.locals`.

#228 is similar, but this is regarding the server fetched data and not _static_ pages.

# Goals

Provide function(s) to refresh data that was server rendered by referencing either:
- The fetch path you want to revalidate
- The tags that have been associated with the request
- The page route that is associated with the request

It is important to be able to reference the request indirectly, since the data may have been loaded from a layout or from the middleware (#469). I would also appreciate it to work in tandem with #532.

# Examples / Inspiration

https://nuxt.com/docs/getting-started/data-fetching#refresh-and-execute
https://kit.svelte.dev/docs/load#rerunning-load-functions-manual-invalidation
https://nextjs.org/docs/app/building-your-application/data-fetching/revalidating#using-on-demand-revalidation

**Comment by lilnasy** [comment]
This is an important issue! Thanks for bringing it up.

**Reply by kyjus25** [reply]
Thanks for the comment, I'm glad you think so too! I think it's necessary for the complete SSR experience, and it follows close to Astro's "everything comes from the server, no JS" philosophy

**Comment by ryands17** [comment]
Is this possible right now?
